Framework/Geometry/src/Instrument.cpp

```diff
@@ -1299,54 +1372,80 @@ boost::shared_ptr<ParameterMap> Instrument::makeLegacyParameterMap() const {
   constexpr double safety_factor = 2.0;
   const double imag_norm_max = sin(d_max / (2.0 * L * safety_factor));
 
-  const IComponent *oldParent{nullptr};
-  Eigen::Quaterniond invParentRot;
   Eigen::Affine3d transformation;
-  for (size_t i = 0; i < m_detectorInfo->size(); ++i) {
-    const auto &det = std::get<1>(baseInstr.m_detectorCache[i]);
-    if (m_detectorInfo->isMasked(i)) {
-      pmap->forceUnsafeSetMasked(det.get(), true);
-    }
 
-    // Obtain parent position/rotation/scale.
-    const auto parent = det->getParent();
-    if (parent.get() != oldParent) {
```

Comment:
Yes. I think this same functionality could be reintroduced just by taking the parent index and comparing it to the last old one.